src/PackageServiceProvider.php

```diff
@@ -63,9 +63,16 @@ public function boot()
             }
 
             if ($this->package->hasViews) {
-                $this->publishes([
-                    $this->package->basePath('/../resources/views') => base_path("resources/views/vendor/{$this->package->shortName()}"),
-                ], "{$this->package->shortName()}-views");
+                if ($this->package->viewNamespace) {
+                    $this->publishes([
+                        $this->package->basePath('/../resources/views') => base_path("resources/views/vendor/{$this->package->viewNamespace}"),
+                    ], "{$this->package->viewNamespace}-views");
+                } else {
```

Comment:
Could you avoid having to use else by extracting this to a function?

Could you also add tests for this?